File: neetcode/7. trees/27.findBottomLeftTreeValue.cpp
// Given the root of a binary tree, return the leftmost value in the last row of the tree. 

#include <limits.h>

struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right): val(x), left(left), right(right) {}
};

class Solution {
public:
  int findBottomLeftValue(TreeNode* root) {
    int res = root->val;
    dfs(root, 0, res);
    return res;
  }
private:
  void dfs(TreeNode* root, int level, int& res) {
    if (root == NULL) {
      return;
    }
    if (level > maxLevel) {
      maxLevel = level;
      res = root->val;
    }
    dfs(root->left, level + 1, res);
    dfs(root->right, level + 1, res);
  }
  int maxLevel = -1;
};

File: neetcode/7. trees/27.findBottomLeftTreeValue.py
# Given the root of a binary tree, return the leftmost value in the last row of the tree. 

class TreeNode:
  def __init__(self, val=0, left=None, right=None):
    self.val = val
    self.left = left
    self.right = right

class Solution:
  def findBottomLeftValue(self, root: TreeNode) -> int:
    if not root:
      return None
    queue = [root]
    while queue:
      node = queue.pop(0)
      if node.right:
        queue.append(node.right)
      if node.left:
        queue.append(node.left)
    return node.val